dev/io.openliberty.reporting.internal/src/io/openliberty/reporting/internal/CVEServiceClient.java

```diff
@@ -0,0 +1,155 @@
+
+		String jsonData = buildJsonString(data);
+		JSONObject json = new JSONObject();
+
+		if (!urlLink.startsWith("https")) {
+			throw new MalformedURLException("Invalid protocol, expected https");
+		}
+
+		URL url = new URL(urlLink);
+
+		HttpsURLConnection connection = getConnection(url);
+		if (connection != null) {
+			sendData(connection, jsonData);
+		}
+
+		if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+			Tr.debug(this, tc, jsonData);
+		}
+
+		try (Reader reader = new InputStreamReader(connection.getInputStream(), StandardCharsets.UTF_8)) {
+			json = JSONObject.parse(reader);
+		}
+		return json;
+	}
+
+	/**
+	 * <p>
+	 * Creates a connection with the cloud Service.
+	 * </p>
+	 * 
+	 * @param url URL
+	 * @return
+	 * @throws IOException
+	 */
+	private static HttpsURLConnection getConnection(URL url) throws IOException {
+		HttpsURLConnection connection = null;
+		connection = (HttpsURLConnection) url.openConnection();
+		connection.setRequestMethod("POST");
+		connection.setRequestProperty("Content-Type", "application/json");
+		connection.setRequestProperty("Accept", "application/json");
+		connection.setDoOutput(true);
+
+		return connection;
+	}
+
+	/**
+	 * <p>
+	 * Sends the data collected from the server to the cloud service.
+	 * </p>
+	 * 
+	 * 
+	 * @param connection HttpsURLConnection
+	 * @param jsonData   String
+	 * @throws ConnectException
+	 * @throws IOException
+	 */
+	private static void sendData(HttpsURLConnection connection, String jsonData) throws ConnectException, IOException {
+		try (OutputStream os = connection.getOutputStream()) {
+			byte[] input = jsonData.getBytes(StandardCharsets.UTF_8);
+			os.write(input, 0, input.length);
+		}
+	}
+
+	/**
+	 * <p>
+	 * Builds the data collected as a string (JSONString) to send.
+	 * </p>
+	 * 
+	 * <pre>
+	 * 	Example: 
+	 * 			 {"productEdition": "edition", "features": ["feature1", "feature2", "feature3"], "productVersion": "12.3.4.56", "iFixes": ["ifix1", "ifix2", "ifix3"], "javaVersion": "17.0.8+7", "id": "STRING", "javaVendor": "javaVendor"}
+	 * </pre>
+	 * 
+	 * 
+	 * 
+	 * @param data A Map<String, String>
+	 * @return A string
+	 */
+	protected String buildJsonString(Map<String, String> data) {
+		StringBuilder jsonData = new StringBuilder("{");
+		for (String key : data.keySet()) {
+			if ("features".equals(key) || "iFixes".equals(key)) {
+				jsonData.append("\"").append(key).append("\": [");
+				if (data.get(key).length() > 0) {
+					String[] featuresOrIFixes = data.get(key).split(",");
+					for (String featureOrIfix : featuresOrIFixes) {
+						jsonData.append("\"").append(featureOrIfix).append("\", ");
+					}
+
+					jsonData.setLength(jsonData.length() - 2);
+				}
+
+				jsonData.append("], ");
+
+			} else {
+				jsonData.append("\"").append(key).append("\": \"").append(data.get(key)).append("\", ");
+			}
+		}
+		jsonData.setLength(jsonData.length() - 2);
```

Comment:
The code assumes that data is not an empty Map.  It appears that that assumption is valid from the callers of the code, but if you want to make the code more robust you may want to add a `if (!data.isEmpty())` around the setLength method or have the first thing in the method be:

```
		if (data.isEmpty()) {
			return "{}";
		}
```